src/backend/winit.rs

```diff
@@ -472,7 +485,35 @@ impl BackendEvent<WinitInputBackend> for WinitMouseInputEvent {
 
 impl PointerButtonEvent<WinitInputBackend> for WinitMouseInputEvent {
     fn button(&self) -> MouseButton {
-        self.button.into()
+        // Wayland and X11 use different mouse button codes on each platform, so we need to know
+        // the platform to properly handle Forward/Back
+        if self.is_x11 {
+            match self.button {
+                WinitMouseButton::Other(raw) if raw == 8 || raw == 9 => {
+                    // X11 is 8 for back and 9 for forward.
+                    match raw {
+                        8 => MouseButton::Back,
+                        9 => MouseButton::Forward,
+                        _ => unreachable!(),
+                    }
+                }
+
+                b => b.into(),
+            }
```

Comment:
We can flatten this match with patterns like `WinitMouseButton::Other(8) => MouseButton::Back`, same for the second one with Wayland.